### ISSUE TITLE
Called cc.game.restart and immediately rotating the screen will cause a crash

### ISSUE THREAD

**Issue by xianyinchen** [cocos/cocos-engine#11712]
### Cocos Creator version

3.4.2

### System information

Android 11

### Issue description

As I know, the EventDispatcher hold a static jsOrientationEventObj  value, when engine restart the v8 will recreate, if orientation event hapend and jsOrientationEventObj  is an old value, that must cause crash.

```c++
void EventDispatcher::dispatchOrientationChangeEvent(int orientation) {
    if (!se::ScriptEngine::getInstance()->isValid()) {
        return;
    }

    se::AutoHandleScope scope;
    assert(inited);

    if (jsOrientationEventObj == nullptr) {
        jsOrientationEventObj = se::Object::createPlainObject();
        jsOrientationEventObj->root();
    }

    se::Value func;
    __jsbObj->getProperty("onOrientationChanged", &func);
    if (func.isObject() && func.toObject()->isFunction()) {
        jsOrientationEventObj->setProperty("orientation", se::Value(orientation));

        se::ValueArray args;
        args.emplace_back(se::Value(jsOrientationEventObj));
        func.toObject()->call(args, nullptr);
    }
}
```

**crash statck:**

![image](https://user-images.githubusercontent.com/8321365/175874709-8fd87800-9dc4-44d1-aae5-253ec3e8ead6.png)

### Steps to reproduce

change the code and always create a new js object, crash will fixed.

```c++
void EventDispatcher::dispatchOrientationChangeEvent(int orientation) {
    if (!se::ScriptEngine::getInstance()->isValid()) {
        return;
    }

    se::AutoHandleScope scope;
    assert(inited);

    jsOrientationEventObj = se::Object::createPlainObject();
    jsOrientationEventObj->root();

    se::Value func;
    __jsbObj->getProperty("onOrientationChanged", &func);
    if (func.isObject() && func.toObject()->isFunction()) {
        jsOrientationEventObj->setProperty("orientation", se::Value(orientation));

        se::ValueArray args;
        args.emplace_back(se::Value(jsOrientationEventObj));
        func.toObject()->call(args, nullptr);
    }

    jsOrientationEventObj->unroot();
    jsOrientationEventObj = nullptr;
}
```

**Comment by minggo** [comment]
@bofeng-song please check if it is fixed on v3.6.

**Comment by bofeng-song** [comment]
fix at this issue:   https://github.com/cocos/cocos-engine/pull/11914
